File: pulse-kotlin/src/commonMain/kotlin/com/fluxchess/pulse/kotlin/engine/CastlingType.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin.engine

typealias CastlingType = Int

const val KINGSIDE: CastlingType = 0
const val QUEENSIDE: CastlingType = 1

const val NO_CASTLING_TYPE: CastlingType = 2

val castlingTypes = intArrayOf(KINGSIDE, QUEENSIDE)

fun isValidCastlingType(castlingType: CastlingType): Boolean = when (castlingType) {
	KINGSIDE, QUEENSIDE -> true
	else -> false
}

File: pulse-go/internal/pulse/engine/castling_type.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package engine

type CastlingType = int

const (
	Kingside  CastlingType = 0
	Queenside CastlingType = 1

	NoCastlingType CastlingType = 2
)

var (
	CastlingTypes = [2]CastlingType{Kingside, Queenside}
)

func IsValidCastlingType(ct CastlingType) bool {
	switch ct {
	case Kingside, Queenside:
		return true
	default:
		return false
	}
}
